aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -166,6 +167,38 @@ static void max_values_kernel_impl(TensorIterator& iter) {
   });
 }
 
+// Maximum and minimum possible scalar values, including infinities
+
+template <typename scalar_t>
+constexpr scalar_t upper_bound() {
```

Comment:
Do you have to define upper/lower bound here because otherwise they are available for cuda only?

Reply:
Yes, `at::numeric_limits` also defines `upper_bound` but only on cuda.